src/mono/mono/component/debugger-agent.c

```diff
@@ -8494,14 +8495,17 @@ method_commands_internal (int command, MonoMethod *method, MonoDomain *domain, g
 			const char *srcfile = "";
 
 			if (source_files [i] != -1) {
-				MonoDebugSourceInfo *sinfo = (MonoDebugSourceInfo *)g_ptr_array_index (source_file_list, source_files [i]);
+				int idx = i;
+				if (i >= n_il_offsets_original)
+					idx = 0;
+				MonoDebugSourceInfo *sinfo = (MonoDebugSourceInfo *)g_ptr_array_index (source_file_list, source_files [idx]);
 				srcfile = sinfo->source_file;
 			}
 			PRINT_DEBUG_MSG (10, "IL%x -> %s:%d %d %d %d\n", sp->il_offset, srcfile, sp->line, sp->column, sp->end_line, sp->end_column);
 			buffer_add_int (buf, sp->il_offset);
 			buffer_add_int (buf, sp->line);
 			if (CHECK_PROTOCOL_VERSION (2, 13))
-				buffer_add_int (buf, source_files [i]);
+				buffer_add_int (buf, i >= n_il_offsets_original ? source_files [0] : source_files [i]);
```

Comment:
can `source_files [0]` ever be -1 here?  Also why is this ok?

Reply:
This is because as I understand on EnC pdb we don't have information about source code.
And we can have more il_offsets then the original pdb, like when you add more lines, so I get the source of the first breakpoint.

Reply:
So `source_files[]` is an array of indexes of the source documents for a particular method? And each `source_files[x]` says where sequence point number x comes from?  So this is saying "if we got additional sequence points beyond the original ones, assume they're from the same document as the first original sequence point?

I guess that makes sense (at least until we have to deal with `partial` C# methods - although I'm not sure if those have sequence points in multiple files).

What if the original method body didn't have any sequence points?  Can that happen?  What if the method was completely empty at the start?  I guess there's still a `ret` in there - and maybe we have a sequence point?

Will debugger-libs be ok if we send back `source_files[0]` and it is `-1`?